src/derivative.c

```diff
@@ -170,20 +185,24 @@ OSQPInt adjoint_derivative(OSQPSolver*    solver,
     OSQPMatrix_free(A_ineq_u);
 
     OSQPVectori* A_eq_i = OSQPVectori_new(A_eq_vec, m);
+    c_free(A_eq_vec);
     OSQPMatrix*  A_eq = OSQPMatrix_submatrix_byrows(A, A_eq_i);
+    OSQPVectori_free(A_eq_i);
+    OSQPMatrix_free(A);
 
     // --------- lambda
     OSQPVectorf* m_zeros = OSQPVectorf_malloc(m);
     OSQPVectorf_set_scalar(m_zeros, 0);
-    OSQPVectorf* y_u = OSQPVectorf_malloc(m);
-    OSQPVectorf_ew_max_vec(y_u, y, m_zeros);
-    OSQPVectorf* y_l = OSQPVectorf_malloc(m);
-    OSQPVectorf_ew_min_vec(y_l, y, m_zeros);
-    OSQPVectorf_mult_scalar(y_l, -1);
+    derivative_data->y_u = OSQPVectorf_malloc(m);
```

Comment:
Since `m` is known at setup time, we should allocate these inside `osqp_setup` alongside the overall `derivative_data` struct. Anything we free in `osqp_cleanup` should be allocated by setup for better traceability (and to maintain consistent code style).

Reply:
done